core/arch/instr_inline.h

```diff
@@ -215,8 +215,8 @@ opnd_create_reg_partial(reg_id_t r, opnd_size_t subsize)
 {
     opnd_t opnd DR_IF_DEBUG(= { 0 }); /* FIXME: Needed until i#417 is fixed. */
 #    ifdef X86
-    CLIENT_ASSERT((r >= DR_REG_MM0 && r <= DR_REG_XMM15) ||
-                      (r >= DR_REG_YMM0 && r <= DR_REG_YMM15),
+    CLIENT_ASSERT((r >= DR_REG_MM0 && r <= DR_REG_XMM31) ||
```

Comment:
This doesn't look right: we'd add new values at the end so the old range here would never change.

Reply:
MMX and XMM enums are contiguous (in new proposed patch that will follow).